api/agent/drivers/docker/docker.go

```diff
@@ -102,13 +164,41 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	liopts := docker.ListImagesOptions{All: false}
+	requiredImages := ListOfImageIDs(conf.RequiredImages)
+	liopts.Context = context.Background()
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
 		if err != nil {
 			logrus.WithError(err).Fatalf("cannot load docker images in %s", conf.DockerLoadFile)
 		}
 	}
 
+	if conf.MaxImageCacheSize != 0 {
+		driver.imageCache = NewCache()
+
+		go func(context context.Context) {
+			liopts := docker.ListImagesOptions{All: false}
+			liopts.Context = context
+			images, err := driver.docker.ListImages(liopts)
+
+			if err != nil {
+				logrus.WithError(err).Fatalf("cannot list docker images %s", err)
+			}
+			for _, i := range images {
+				if Contains(requiredImages, i) {
+					driver.imageCache.Add(i)
+				} else {
+					driver.imageCache.Add(i)
+					driver.imageCache.Lock(i.ID, "baseimage")
+				}
+			}
+		}(context.Background())
```

Comment:
any chance we can manage these 2 goroutines with the driver.Close() method? for testing this could get out of hand if we launch these but never shut them down. nitty but would be nice to extract this one to a separate function to call too out of NewDocker()